astropy/table/groups.py

```diff
@@ -54,16 +85,25 @@ def _table_group_by(table, keys):
         if len(table_keys) != len(table):
             raise ValueError('Input keys array length {0} does not match table length {1}'
                              .format(len(table_keys), len(table)))
+
         grouped_by_table_cols = False  # Grouping key(s) are external
 
     else:
         raise TypeError('Keys input must be string, list, tuple or numpy array, but got {0}'
                         .format(type(keys)))
 
+    # If table_keys is a Table then ensure that all cols (including mixins) are
+    # in a form that can sorted with the code below.
+    if isinstance(table_keys, Table):
```

Comment:
I just realized that the logic here will stomp on a Time column which is indexed, and prevent the optimization below of using the already-existing index to sort.  So that needs to be fixed.  Testing is slightly tricky because you get the same answer either way, just one is faster.